hikari/impl/event_manager.py

```diff
@@ -162,50 +157,55 @@ async def on_channel_pins_update(self, shard: gateway_shard.GatewayShard, payloa
         # TODO: we need a method for this specifically
         await self.dispatch(self._event_factory.deserialize_channel_pins_update_event(shard, payload))
 
-    # Internal granularity is preferred for GUILD_CREATE over decorator based filtering due to its large cache scope.
-    async def on_guild_create(self, shard: gateway_shard.GatewayShard, payload: data_binding.JSONObject) -> None:
+    @event_manager_base.filtered(
+        (guild_events.GuildAvailableEvent, guild_events.GuildJoinEvent),
+        config.CacheComponents.GUILDS
+        | config.CacheComponents.GUILD_CHANNELS
```

Comment:
What's the point of checking it twice and since the comment was removed is the internal granularity also going to be removed?

Reply:
You also don't account for chunking here and won't be able to if an explicit chunk setting is added so...

Reply:
The check is really fast, so I didn't see the point of having it dispatch every time, regardless of the cache settings/listeners/waiters